search_service/proxy/atlas.py

```diff
@@ -172,82 +286,62 @@ def fetch_table_search_results_with_field(self, *,
 
-        # noinspection PyProtectedMember
-        tables_count = table_results._data.get("approximateCount")
-
-        tables = []
-        for table in table_results.entities:
-            table_attrs = table.attributes
-
-            table_qn = parse_table_qualified_name(
-                qualified_name=table_attrs.get(self.QN_KEY)
-            )
-
-            table_name = table_qn.get("table_name") or table_attrs.get('name')
-            db_name = table_qn.get("db_name", '')
-            db_cluster = table_qn.get("cluster_name", '')
-            table = Table(name=table_name,
-                          key=f"{table.typeName}://{db_cluster}.{db_name}/{table_name}",
-                          description=table_attrs.get('description') or table_attrs.get('comment'),
-                          cluster=db_cluster,
-                          database=table.typeName,
-                          schema=db_name,
-                          column_names=[],
-                          tags=[],
-                          badges=[],
-                          last_updated_timestamp=table_attrs.get('updateTime'))
+        filters = list()
 
-            tables.append(table)
+        if db_filter_value:
+            filters.append((self.ATLAS_QN_ATTRIBUTE, 'STARTSWITH', db_filter_value[0] + '.'))
 
-        return tables, tables_count
+        if cluster_filter_value:
+            filters.append((self.ATLAS_QN_ATTRIBUTE, 'ENDSWITH', '@' + cluster_filter_value[0]))
 
-    @timer_with_counter
-    def fetch_table_search_results(self, *,
-                                   query_term: str,
-                                   page_index: int = 0,
-                                   index: str = '') -> SearchResult:
-        """
-        Query Atlas and return results as list of Table objects.
-        Using Basic Search for this basic searching.
+        if table_filter_value:
+            filters.append(('name', 'CONTAINS', table_filter_value[0]))
 
-        :param query_term: search query term
-        :param page_index: index of search page user is currently on
-        :param index: search index (different resource corresponding to different index)
-        :return: SearchResult Object
-        """
-        if not query_term:
-            # return empty result for blank query term
-            return SearchResult(total_results=0, results=[])
+        classifications: List[str] = list()  # noqa: E701
+
+        if badges_filter_value or tags_filter_value:
+            classifications = list(set(badges_filter_value + tags_filter_value))
 
-        query_params = {'typeName': 'Table',
-                        'excludeDeletedEntities': True,
-                        'limit': self.page_size,
-                        'offset': page_index * self.page_size,
-                        'query': f'*{query_term}*',
-                        'attributes': ['description', 'comment']}
+        # Currently Atlas doesn't allow mixing search by filters and classifications
```

Comment:
Hmmm, good catch. Might like to add it to the doc comment too and in mention this limitation in the doc comment for the called method so that future callsites can take care.